js/main.js

```diff
@@ -90,56 +102,112 @@ map.classList.add('map--faded');
   adForm.classList.remove('ad-form--disabled');
-  disableAdForm();
-  disableFiltersForm();
+  ableAdForm();
+  ableFiltersForm();
   renderPins(MOKS_OBJECT_NUMBERS);
 };
 
-// Обработчик отжатия клика на основной метке
-var mapPinMainMouseupHandler = function () {
+// Вносит коорднаты метки в поле адреса
+var setAddress = function () {
   address.setAttribute('value', getPinMainCoords());
 };
 
-// Активация карты и форм по нажатию на основную метку
-mapPinMain.addEventListener('click', mapPinMainClickHandler);
-
-// Функция определения координаты основной метки
+// Возвращает координаты острого конца метки
 var getPinMainCoords = function () {
   var xPinCoord = parseInt(mapPinMain.style.left, 10) + MAP_PIN_MAIN_WIDTH / 2;
   var yPinCoord = parseInt(mapPinMain.style.top, 10) + MAP_PIN_MAIN_HEIGHT;
   return (xPinCoord + ', ' + yPinCoord);
 };
 
-// Определение координат метки после нажатия
-mapPinMain.addEventListener('mouseup', mapPinMainMouseupHandler);
+// Перемещение метки
+mapPinMain.addEventListener('mousedown', function (evt) {
+  evt.preventDefault();
+  activateForms();
+
+  var startCoords = {
+    x: evt.clientX,
+    y: evt.clientY
+  };
+
+  // Обработчик перемещения
+  var mouseMoveHandler = function (moveEvt) {
+    moveEvt.preventDefault();
+
+    // Величина смещения по осям
+    var shift = {
+      x: moveEvt.clientX - startCoords.x,
+      y: moveEvt.clientY - startCoords.y
+    };
+
+    // Расположение метки в окне без [px]
+    var pinMargin = {
+      top: parseInt(mapPinMain.style.top, 10),
+      left: parseInt(mapPinMain.style.left, 10)
+    };
+
+    startCoords = {
+      x: moveEvt.clientX,
+      y: moveEvt.clientY
+    };
+
+    // Координаты после смещения
+    mapPinMain.style.top = (mapPinMain.offsetTop + shift.y) + 'px';
+    mapPinMain.style.left = (mapPinMain.offsetLeft + shift.x) + 'px';
+
+    // Наложение ограничений по границам окна map__pins
+    if (pinMargin.top < mapPins.offsetTop + MAP_TOP_BORDER) {
+      mapPinMain.style.top = mapPins.offsetTop + MAP_TOP_BORDER + 'px';
+    }
+    if (pinMargin.top > mapPins.offsetHeight - filterFormStyleHeight) {
+      mapPinMain.style.top = mapPins.offsetHeight - filterFormStyleHeight + 'px';
+    }
+    if (pinMargin.left < mapPins.offsetLeft) {
+      mapPinMain.style.left = mapPins.offsetLeft + 'px';
+    }
+    if (pinMargin.left > mapPins.offsetWidth - MAP_PIN_MAIN_WIDTH) {
+      mapPinMain.style.left = mapPins.offsetWidth - MAP_PIN_MAIN_WIDTH + 'px';
+    }
+  };
+
+  var mouseUpHandler = function (upEvt) {
+    upEvt.preventDefault();
+    setAddress();
+
+    document.removeEventListener('mousemove', mouseMoveHandler);
+    document.removeEventListener('mouseup', mouseUpHandler);
+  };
+
+  document.addEventListener('mousemove', mouseMoveHandler);
+  document.addEventListener('mouseup', mouseUpHandler);
+});
 
 // Обработчик опций выбора места
 var setMinPriceHandler = function (placeType) {
   var priceValue = 0;
```

Comment:
в таком случае switch/case не нужен

```suggestion
  var priceValue = PRICES[placeType];
```